packages/radix-ui-themes/src/theme.tsx

```diff
@@ -176,12 +142,9 @@ const ThemeImpl = React.forwardRef<ThemeImplElement, ThemeImplProps>((props, for
   } = props;
   const Comp = asChild ? Slot : 'div';
   const resolvedGrayColor = grayColor === 'auto' ? getMatchingGrayColor(accentColor) : grayColor;
-  const isExplicitAppearance = props.appearance !== undefined && props.appearance !== 'inherit';
-  const isExplicitGrayColor = props.grayColor !== undefined;
-  const shouldHaveBackground =
-    !isRoot &&
-    (hasBackground === true ||
-      (hasBackground !== false && (isExplicitAppearance || isExplicitGrayColor)));
```

Comment:
We had some situations in AuthKit when passing an explicit `hasBackground` didn't have any effect

I also removed `isExplicitGrayColor` condition to simplify the logic as it was unintuitive. I recollect that the argument was to set background color when the gray tint is changed so that is visible in dark mode, but imo it's too fringe vs the complexity we are getting.